cpp/include/cuspatial/hausdorff.hpp

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2019-2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <memory>
+#include <cudf/types.h>
+
+namespace cuspatial {
+
+/**
+ * @brief computes Hausdorff distances for all pairs in a collection of spaces
+ *
+ * https://en.wikipedia.org/wiki/Hausdorff_distance
+ *
+ * Example in 1D (this function operates in 2D):
+ * ```
+ * spaces
+ * [0 2 5] [9] [3 7]
+ *
+ * spaces represented as points per space and concatenation of all points
+ * [0 2 5 9 3 7] [3 1 2]
+ *
+ * cartesian product of all
+ * points visually seperated
+ * by pair of spaces            distance between points
```

Comment:
"visually separated by pair of spaces" is confusing. Do you mean the character "space"? The table on the right has two spaces separating each entry -- is that the visual separation you mean?

Reply:
What about “visually separated to highlight the relationship of a pair of points with the pair of spaces from which it was produced” ?

Wordy, but maybe less ambiguous?

Reply:
Maybe labeling the sets and using those labels at the top and sides of the cells could make sense?

Reply:
```
 * note: the following matricies are visually separated to highlight the relationship of a pair of
 * points with the pair of spaces from which it is produced
 * 
 * cartesian product of all
 * points by pair of spaces     distance between points
 * +----------+----+-------+    +---------+---+------+
```